File: qa_verify.py
from playwright.sync_api import sync_playwright
import time
import json

def verify_env(url):
    results = {}
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        
        # 1. Onboarding Flow
        try:
            context = browser.new_context()
            page = context.new_page()
            page.goto(f"{url}/details")
            # Submit empty to trigger validation
            page.click("button:has-text('Get My Strategy')", timeout=5000)
            
            # Check for error state
            error_el = page.locator(".text-red-500").first
            error_text = error_el.inner_text() if error_el.count() > 0 else "No error text found"
            
            # Check if inline validation exists
            has_validation = "No error text found" not in error_text
            results['onboarding'] = {"has_validation": has_validation, "error_text": error_text}
            context.close()
        except Exception as e:
            results['onboarding'] = {"error": str(e)}

        # 2. Admin Knowledge Library Pagination
        try:
            context = browser.new_context()
            page = context.new_page()
            page.goto(f"{url}/admin/knowledge")
            page.wait_for_selector("table", timeout=5000)
            
            # Look for pagination buttons (Next, Previous)
            has_pagination = page.locator("button:has-text('Next')").count() > 0
            has_search = page.locator("input[placeholder*='Search']").count() > 0
            
            results['admin_library'] = {"has_pagination": has_pagination, "has_search": has_search}
            context.close()
        except Exception as e:
            results['admin_library'] = {"error": str(e)}

        # 3. Knowledge Editor JSON Validation
        try:
            context = browser.new_context()
            page = context.new_page()
            page.goto(f"{url}/admin/knowledge/edit") # Might redirect if no ID, but let's see
            time.sleep(2)
            has_change_summary = page.locator("input[placeholder*='Summary']").count() > 0 or page.locator("textarea[placeholder*='Summary']").count() > 0
            
            results['admin_editor'] = {"has_change_summary": has_change_summary}
            context.close()
        except Exception as e:
            results['admin_editor'] = {"error": str(e)}

        browser.close()
    return results

print("=== LOCAL (http://localhost:8788) ===")
local_results = verify_env("http://localhost:8788")
print(json.dumps(local_results, indent=2))

print("\n=== PRODUCTION (https://main.cac-web-app.pages.dev) ===")
prod_results = verify_env("https://main.cac-web-app.pages.dev")
print(json.dumps(prod_results, indent=2))

File: qa_verify.js
const { chromium } = require('playwright');
const fs = require('fs');

async function verifyEnv(url) {
    const results = {};
    const browser = await chromium.launch({ headless: true });
    
    try {
        // 1. Onboarding Flow
        try {
            const context = await browser.newContext();
            const page = await context.newPage();
            await page.goto(`${url}/details`);
            
            // Wait for button and click
            await page.waitForSelector('button:has-text("Get My Strategy")');
            await page.click('button:has-text("Get My Strategy")');
            
            // Check for error state
            await page.waitForTimeout(1000); // give it a sec to show validation
            const errorEls = await page.$$('.text-red-500');
            let errorText = 'No error text found';
            if (errorEls.length > 0) {
                errorText = await errorEls[0].innerText();
            }
            
            results.onboarding = { 
                has_validation: errorText !== 'No error text found', 
                error_text: errorText 
            };
            await context.close();
        } catch (e) {
            results.onboarding = { error: e.message };
        }

        // 2. Admin Knowledge Library Pagination
        try {
            const context = await browser.newContext();
            const page = await context.newPage();
            await page.goto(`${url}/admin/knowledge`);
            await page.waitForSelector('table', { timeout: 5000 });
            
            const nextBtns = await page.$$('button:has-text("Next")');
            const searchInputs = await page.$$('input[placeholder*="Search"]');
            
            results.admin_library = { 
                has_pagination: nextBtns.length > 0, 
                has_search: searchInputs.length > 0 
            };
            await context.close();
        } catch (e) {
            results.admin_library = { error: e.message };
        }

        // 3. Knowledge Editor
        try {
            const context = await browser.newContext();
            const page = await context.newPage();
            // Goto edit page for a dummy ID so we can see the form
            await page.goto(`${url}/admin/knowledge/edit?id=new`);
            await page.waitForTimeout(2000);
            
            const summaryInputs = await page.$$('input[placeholder*="Summary"], textarea[placeholder*="Summary"], input[id="change-summary"]');
            const hasChangeSummary = summaryInputs.length > 0;
            
            results.admin_editor = { has_change_summary: hasChangeSummary };
            await context.close();
        } catch (e) {
            results.admin_editor = { error: e.message };
        }
    } finally {
        await browser.close();
    }
    return results;
}

(async () => {
    console.log("=== LOCAL (http://localhost:8788) ===");
    const local = await verifyEnv("http://localhost:8788");
    console.log(JSON.stringify(local, null, 2));

    console.log("\n=== PRODUCTION (https://main.cac-web-app.pages.dev) ===");
    const prod = await verifyEnv("https://main.cac-web-app.pages.dev");
    console.log(JSON.stringify(prod, null, 2));
})();
